packages/grid/test/lit.common.js

```diff
@@ -58,4 +59,75 @@ describe('lit', () => {
     // Then render a grid with one column
     renderGrid(['c1'], [{ c1: '1-1', c2: '1-2' }]);
   });
+
+  // Test case for https://github.com/vaadin/web-components/issues/7162
+  it('should render child item content correctly', async () => {
+    const testComponentName = 'test-component';
+
+    // Define a web component that flushes global debouncers on connect
+    class TestComponent extends HTMLElement {
+      constructor() {
+        super();
+        this.attachShadow({ mode: 'open' }).innerHTML = '<slot></slot>';
+      }
+
+      /** @protected */
+      connectedCallback() {
+        this.innerHTML = 'Foobar';
+        flush();
+      }
+    }
+    customElements.define(testComponentName, TestComponent);
+
+    // Build items
+    const rootItem = { id: 'item', children: [] };
+    for (let i = 0; i < 3; i++) {
+      const item = { id: `item_${i}`, children: [] };
+      rootItem.children.push(item);
+      for (let j = 0; j < 1; j++) {
+        const childItem = { id: `item_${i}_${j}` };
+        item.children.push(childItem);
+      }
+    }
+
+    const expandedItems = [rootItem, rootItem.children[1]];
+
+    const dataProvider = (params, callback) => {
+      const items = params.parentItem?.children || [rootItem];
+      callback(items, items.length);
+    };
+
+    const renderer = (root, _column, { item }) => {
+      let itemContent;
+
+      if (item.children) {
+        itemContent = html`${item.id}`;
+      } else {
+        itemContent = document.createElement('test-component');
+      }
+
+      render(html`<div>${itemContent}</div>`, root);
+    };
+
+    const wrapper = fixtureSync(`<div></div>`);
+    render(
```

Comment:
Is using Lit essential to reproduce the target issue?

Reply:
Use of Lit is required in the renderer function to reproduce the issue.

The grid itself could be rendered with any means in this case, but Lit is used for convenience and to align with the other tests in this test file.